File: epikjjh/baekjoon/1260.py
n, m, start = map(int, input().split())
matrix = [[0]*(n+1) for i in range(n+1)]
flag = [0]*(n+1)
queue = [start]
for i in range(m):
    a, b = map(int, input().split())
    matrix[a][b] = matrix[b][a] = 1


def dfs(vertex):
    if not flag[vertex]:
        print(vertex, end=" ")
        flag[vertex] = 1
        for i in range(1, n+1):
            if matrix[vertex][i]:
                dfs(i)


dfs(start)
print()
flag = [0]*(n+1)
while queue:
    vertex = queue.pop(0)
    if not flag[vertex]:
        print(vertex, end=" ")
        flag[vertex] = 1
        for i in range(1, n+1):
            if matrix[vertex][i]:
                queue.append(i)
print()

File: epikjjh/baekjoon/1260.cpp
#include <bits/stdc++.h>

using namespace std;

vector<vector<int>> adj;
bool visited[1001];

void dfs(int cur){
    visited[cur] = true;
    printf("%d ",cur);
    for(int nxt: adj[cur])  if(!visited[nxt])   dfs(nxt);
}

int main(){
    int n,m,v;
    scanf("%d %d %d",&n,&m,&v);
    adj.resize(n+1);
    while(m--){
        int a,b;
        scanf("%d %d",&a,&b);
        adj[a].push_back(b);
        adj[b].push_back(a);
    }
    for(int i=1;i<=n;i++)   sort(adj[i].begin(),adj[i].end());
    dfs(v);
    printf("\n");
    memset(visited,0,sizeof(visited));
    queue<int> q;
    q.push(v);
    visited[v] = true;
    while(!q.empty()){
        int cur = q.front();
        q.pop();
        printf("%d ",cur);
        for(int nxt: adj[cur]){
            if(!visited[nxt]){
                visited[nxt] = true;
                q.push(nxt);
            }
        }
    }
    printf("\n");

    return 0;
}
